spec/models/course_spec.rb

```diff
@@ -434,4 +434,103 @@
       its(:content_status) { should eq(:published_with_unpublished_changes) }
     end
   end
+
+  describe "#publish_enrichments" do
+    context 'on a course with only a draft enrichment' do
+      let(:course) do
+        create(:course,
+            changed_at: 10.minutes.ago,
+          with_enrichments: [
+          [:initial_draft, created_at: 1.day.ago, updated_at: 20.minutes.ago],
+        ])
+      end
+      let(:user) { create(:user) }
+
+      it 'should publish the draft' do
+        course.publish_enrichment(user.id)
+        expect(course.enrichments.first.status).to eq 'published'
+      end
+
+      it 'should update course changed_at to the current time' do
+        course.publish_enrichment(user.id)
+        expect(course.changed_at).to be_within(1.second).of Time.now.utc
+      end
+
+      it 'should touch enrichment updated_at to the current time' do
+        course.publish_enrichment(user.id)
+        expect(course.enrichments.first.updated_at).to be_within(1.second).of Time.now.utc
+      end
+
+      it 'should updated last_published to the current time' do
+        course.publish_enrichment(user.id)
+        expect(course.enrichments.first.last_published_timestamp_utc).to be_within(1.second).of Time.now.utc
+      end
+
+      it 'should updated updated_by to the current user' do
+        course.publish_enrichment(user.id)
+        expect(course.enrichments.first.updated_by_user_id).to eq user.id
+      end
+    end
+
+    context 'on a course draft and published enrichments' do
```

Comment:
Maybe: on a course with a draft enrichment and previously-published

Whatever it may be, I think a couple more words in here will help. :)

Reply:
done, thanks